kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1522 @@
+     */
+    void SystemSolveWithPhysics(
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb,
+        ModelPart& rModelPart
+        )
+    {
+        KRATOS_TRY
+
+        double norm_b;
+        if (TSparseSpace::Size(rb) != 0)
+            norm_b = TSparseSpace::TwoNorm(rb);
+        else
+            norm_b = 0.0;
+
+        if (norm_b != 0.0) {
+
+             // Create the auxiliar dof set
+             DofsArrayType aux_dof_set;
+             aux_dof_set.reserve(mDoFToSolveSystemSize);
+             for (auto& dof : BaseType::mDofSet) {
+                 if (dof.EquationId() < BaseType::mEquationSystemSize) {
+                    auto it = mDoFSlaveSet.find(dof);
+                    if (it == mDoFSlaveSet.end())
+                        aux_dof_set.push_back( &dof );
+                 }
+             }
+             aux_dof_set.Sort();
+
+             KRATOS_ERROR_IF_NOT(aux_dof_set.size() == mDoFToSolveSystemSize) << "Inconsistency (I) in system size: " << mDoFToSolveSystemSize << " vs " << aux_dof_set.size() << "\n Size dof set " << BaseType::mDofSet.size() << " vs Size slave dof set " << mDoFSlaveSet.size() << std::endl;
+             KRATOS_ERROR_IF_NOT(aux_dof_set.size() == rA.size1()) << "Inconsistency (II) in system size: " << rA.size1() << " vs " << aux_dof_set.size() << "\n Size dof set " << BaseType::mDofSet.size() << " vs Size slave dof set " << mDoFSlaveSet.size() << std::endl;
+
+            // Provide physical data as needed
+            if(BaseType::mpLinearSystemSolver->AdditionalPhysicalDataIsNeeded())
+                BaseType::mpLinearSystemSolver->ProvideAdditionalData(rA, rDx, rb, aux_dof_set, rModelPart);
+
+            // Do solve
+            BaseType::mpLinearSystemSolver->Solve(rA, rDx, rb);
+        } else {
+            TSparseSpace::SetToZero(rDx);
+            KRATOS_WARNING_IF("ResidualBasedEliminationBuilderAndSolver", rModelPart.GetCommunicator().MyPID() == 0) << "ATTENTION! setting the RHS to zero!" << std::endl;
+        }
+
+        // Prints informations about the current time
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", this->GetEchoLevel() > 1 && rModelPart.GetCommunicator().MyPID() == 0) << *(BaseType::mpLinearSystemSolver) << std::endl;
+
+        KRATOS_CATCH("")
+
+    }
+
+    /**
+     * @brief This function is exactly same as the ConstructMatrixStructure() function in base class except that the function
+     * @details Has the call to ApplyConstraints function call once the element and conditions compute their equation ids
+     * @todo Move this method to a common class with block builder and solver with constraints
+    */
+    virtual void ConstructMatrixStructureWithConstraints(
+        typename TSchemeType::Pointer pScheme,
+        TSystemMatrixType& rA,
+        ModelPart& rModelPart
+        )
+    {
+        // Filling with zero the matrix (creating the structure)
+        Timer::Start("MatrixStructure");
```

Comment:
cann't we call the first the base class ConstructMatrixStructure first and then just do the constraints part here ?

Reply:
I think not

Reply:
ok .. !